firmware/raspi/boardinterface.py

```diff
@@ -229,15 +233,24 @@ def send_message_to_arduino(self, board_move):
 
         print(f"Sending message \"{msg}\" to arduino")
 
-        # TODO: Implement sending message to arduino
+        
+        #Serial message from Pi to Arduino
+        ser.write(msg)
 
         # TODO: This is for game loop dev, remove once we read from arduino
         self.set_status_from_arduino(ArduinoStatus.EXECUTING_MOVE, board_move.move_count, None)
 
     def get_status_from_arduino(self):
         '''Read status from Arduino over UART connection.
         '''
-        # TODO: update this function to actually read from arduino
+        #New variable created, new_input, to store 4 bytes for UART Messages
+        #If the start byte is a ~ and the Arduinio Status is valid, process the arduino status based on the new input
+        new_input = ser.read(4)
+        if new_input[0] = '~' and (new_input[1] == ArduinoStatus.IDLE 
+                                or new_input[1] == ArduinoStatus.EXECUTING_MOVE 
+                                or new_input[1] == ArduinoStatus.END_TURN_BUTTON_PRESSED
+                                or new_input[1] == ArduinoStatus.ERROR):
```

Comment:
I would prefer the new_input[1] check to be done in a function in the ArduinoStatus class. Something like
```python
# This function declared in the ArduinoStatus class
def is_valid_code(self, code):
    return code in (ArduinoStatus.IDLE,
                    ArduinoStatus.EXECUTING_MOVE,
                    ArduinoStatus.END_TURN_BUTTON_PRESSED
                    ArduinoStatus.ERROR)
```
and then if that return value is false, the function should raise a `ValueError` letting us know something went wrong
```python
raise ValueError(f"Error: received unexpected status code: {new_input[1]}...")
```